### PR TITLE
fix(pci-private-network): add vrack creation

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | master
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | DTCORE-2764
| License          | BSD 3-Clause

## Description

Add missing vrack creation / association modal on pci-private-networks

### CODE REVIEW

**Review by kqesar — APPROVED** [review]

File: packages/manager/apps/pci-private-network/src/api/data/vrack.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+export const pollOperation = async (
+  projectId: string,
+  operationId: string,
+  intervalMs = 1000,
+) => {
+  const poll = async () => {
+    const { data: op } = await v6.get(
+      `/cloud/project/${projectId}/operation/${operationId}`,
+    );
+    if (op.status === 'completed') {
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-private-network/src/api/data/vrack.ts]: done

File: packages/manager/apps/pci-private-network/src/api/hooks/useVrack.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  createVrack,
+  getProjectVrack,
+  getVracks,
+  pollOperation,
+  pollVrackTask,
+} from '@/api/data/vrack';
+
+export const useProjectVrack = (projectId: string) =>
+  useQuery({
+    queryKey: ['project-vrack'],
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-private-network/src/api/hooks/useVrack.ts]: done

File: packages/manager/apps/pci-private-network/src/pages/onboarding/Onboarding.page.tsx
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-private-network/src/pages/onboarding/Onboarding.page.tsx:1]: added a test case

File: packages/manager/apps/pci-private-network/src/pages/onboarding/new/VrackCreation.page.tsx
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/pci-private-network/src/pages/onboarding/new/VrackCreation.page.tsx:1]: added a snapshot test

**Review by darsene — DISMISSED** [review]

**Review by kqesar — DISMISSED** [review]

**Review by kqesar — APPROVED** [review]

**Review by seven-amid — APPROVED** [review]

**Pull request merged**